components/camel-pulsar/src/main/java/org.apache.camel.component.pulsar/PulsarComponent.java

```diff
@@ -1,24 +1,69 @@
 package org.apache.camel.component.pulsar;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+
 import org.apache.camel.CamelContext;
 import org.apache.camel.Endpoint;
+import org.apache.camel.component.pulsar.configuration.AdminConfiguration;
 import org.apache.camel.component.pulsar.configuration.PulsarEndpointConfiguration;
 import org.apache.camel.impl.DefaultComponent;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
+import org.apache.pulsar.common.policies.data.TenantInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class PulsarComponent extends DefaultComponent {
 
-    public PulsarComponent(CamelContext context) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarComponent.class);
+
+    private final PulsarEndpointConfiguration configuration;
+    private final AdminConfiguration adminConfiguration;
+
+    PulsarComponent(CamelContext context, PulsarEndpointConfiguration configuration, AdminConfiguration adminConfiguration) {
         super(context);
+        this.configuration = configuration;
+        this.adminConfiguration = adminConfiguration;
     }
 
     @Override
-    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
-
-        PulsarEndpointConfiguration configuration = new PulsarEndpointConfiguration(new PulsarUri(uri));
+    protected Endpoint createEndpoint(String uri, String path, Map<String, Object> parameters) throws Exception {
 
         setProperties(configuration, parameters);
 
-        return PulsarEndpoint.create(configuration, configuration.getPulsarClient());
+        ensureNameSpaceAndTenant(path);
```

Comment:
what happens when the admin config file is null and the tenant and namespace is not supplied in the path?

Reply:
Then a topic is created in the default namespace, in the public tenant. This is the default behaviour and if we wish to create a PR upstream must be preserved.